ops/main.py

```diff
@@ -105,22 +121,21 @@ def _get_event_args(charm, bound_event):
         relation_name = os.environ['JUJU_RELATION']
         relation_id = int(os.environ['JUJU_RELATION_ID'].split(':')[-1])
         relation = model.get_relation(relation_name, relation_id)
-    else:
-        relation = None
-
-    remote_app_name = os.environ.get('JUJU_REMOTE_APP', '')
-    remote_unit_name = os.environ.get('JUJU_REMOTE_UNIT', '')
-    if remote_app_name or remote_unit_name:
-        if not remote_app_name:
-            if '/' not in remote_unit_name:
-                raise RuntimeError(f'invalid remote unit name: {remote_unit_name}')
-            remote_app_name = remote_unit_name.split('/')[0]
-        args = [relation, model.get_app(remote_app_name)]
-        if remote_unit_name:
-            args.append(model.get_unit(remote_unit_name))
-        return args, {}
-    elif relation:
-        return [relation], {}
+        remote_app_name = os.environ.get('JUJU_REMOTE_APP', '')
+        remote_unit_name = os.environ.get('JUJU_REMOTE_UNIT', '')
+        if remote_app_name or remote_unit_name:
```

Comment:
This logic changed shape in a way unrelated to functions. The original logic seemed fine, and in either case it seems unrelated to functions so would be better done separately.